### DISCUSSION TITLE
Signing Transaction Offline

### DISCUSSION THREAD

**Discussion by Chevy-Lu** [XRPLF/xrpl.js#2175] [Q&A] [answered]
While attempting to sign a transaction the script requires a seed in a base58 format. 

My question is:

How can I convert my seed into a base58 format in order to sign transactions? 

Thanks in advance!

**Comment by mvadari** [comment, marked as answer]
Your seed should be in the form `s....`. There's a method in ripple-address-codec called `encodeSeed`, which is probably what you're looking for.

**Reply by Chevy-Lu** [reply]
Hi @mvadari 

Thank you so much for your reply. I tried this method but for some reason it only allows for an 8bytes entropy. But when I read the source code it says 16bytes. More than likely  I am missing something.

My goal is to use an entropy generated from my ledger wallet with 24 words + passphrase and encode the seed into the s... format used by the xrpl so I can submit transactions through my rippled node. 

Problem: I don't think the encodeSeed() from the 'ripple-address-codec' can be able to use the entropy generated from the ledger wallet due to it being more than 16bytes.

Can I change the bytes size limit of encodeSeed() to accommodate my entropy?

Would it still be compatible with the xrpl network?

Is there any other way to sign transactions besides using the seed in the s... format?

Can I use a private key instead (if yes please explain)?

Thanks!

**Reply by mvadari** [reply]
You can instantiate a wallet object with `new Wallet(publicKey, privateKey)`.

**Reply by Chevy-Lu** [reply]
I am getting an undefined seed...
(test wallet)

const xrpl = require('xrpl')

const publicKey = '02e5670c72e68c1bbd150ae756628ad9dc1eb63be4be35ac0e9dcd49c9124b5097';
const privateKey = 'd8bd39d436d1070f554aaac34b1de2cf94f77d062b961b3e1daaa1ededcd2e4b';
const seed = '2fb5178a2db517a108edd0996172f4a3109c848d7c987356e0e8c403c0e402de313616dc20b5d25c24569100bd5fa0de40dcbf875cd600b2f120b1249b8b3ddd'

const my_wallet = new xrpl.Wallet(publicKey, privateKey, seed)

console.log(my_wallet);

//Output:
/* 
Wallet {
    publicKey: '02e5670c72e68c1bbd150ae756628ad9dc1eb63be4be35ac0e9dcd49c9124b5097',
    privateKey: 'd8bd39d436d1070f554aaac34b1de2cf94f77d062b961b3e1daaa1ededcd2e4b',
    classicAddress: 'rp5qaUkMkh6aLFzjwxXDMEHE5S86T3ERuw',
    seed: undefined
  }
*/

**Reply by mvadari** [reply]
You don't need to seed to sign - you only need the public and private keys.
https://js.xrpl.org/classes/Wallet.html#constructor

**Reply by Chevy-Lu** [reply]
Thank you so much it's been an honor talking to you.
